index.js

```diff
@@ -17,11 +18,24 @@ CoffeeScriptFilter.prototype.targetExtension = 'js'
 CoffeeScriptFilter.prototype.processString = function (string, srcFile) {
   var coffeeScriptOptions = {
     bare: this.bare,
-    literate: coffeeScript.helpers.isLiterate(srcFile)
+    literate: coffeeScript.helpers.isLiterate(srcFile),
+    sourceMap: !!this.sourceMap,
+
+    sourceFiles: [ srcFile ],
+    inline: true
   }
 
   try {
-    return coffeeScript.compile(string, coffeeScriptOptions)
+    var compiled = coffeeScript.compile(string, coffeeScriptOptions)
+    var source = this.sourceMap ? compiled.js : compiled
+    var sourceMap = compiled.v3SourceMap
+
+    if (sourceMap) {
+      source += '\n//# sourceMappingURL=data:application/json;base64,'
+      source += new Buffer(sourceMap).toString('base64')
```

Comment:
So that non-ASCII characters in the source show up correctly in the devtools, we should specify the charset

```js
      source += '\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,'
      source += new Buffer(sourceMap, 'utf8').toString('base64')
```

The `utf8` parameter to `Buffer` is already the default, but it can't hurt to be explicit.